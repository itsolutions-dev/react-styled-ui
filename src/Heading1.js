import styled from 'styled-components';

const Heading1 = styled.h1`
  font-size:28px;
  font-weight: 600;
  line-height:30px;
  color:#5d5c5c;
`;

export default Heading1;
