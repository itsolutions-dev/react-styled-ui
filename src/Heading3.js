import styled, { css } from './styledComponent';
import { getTextColor, getColorWithAlpha } from './utils/';

const Heading3 = styled({
  component: 'h3',
  style: css`
    font-size: ${props => props.theme.h3.fontSize};
    font-weight: ${props => props.theme.h3.fontWeight};
    line-height: ${props => props.theme.h3.lineHeight};
    color: ${props => getColorWithAlpha(getTextColor(props), 0.54)};
  `,
});

export default Heading3;
