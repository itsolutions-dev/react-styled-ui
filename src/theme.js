export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
