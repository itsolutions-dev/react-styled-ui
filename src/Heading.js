// @flow

import React from 'react';
import Heading1 from './Heading1';
import Heading2 from './Heading2';
import Heading3 from './Heading3';
import Heading4 from './Heading4';
import Heading5 from './Heading5';
import Heading6 from './Heading6';

const headings = [
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  Heading5,
  Heading6,
];

type HeadingProps = {
  size: 1 | 2 | 3 | 4 | 5 | 6,
};

const Heading = ({ size = 1, ...others }: HeadingProps) => {
  const Component = headings[size - 1];
  return <Component {...others} />;
};

export default Heading;
