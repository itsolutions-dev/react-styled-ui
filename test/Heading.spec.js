import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import Heading from '../src/Heading';
import Heading1 from '../src/Heading1';
import Heading3 from '../src/Heading3';
import theme from '../src/theme';

describe('Heading', () => {
  test('should render an Heading', () => {
    const component = ReactTestRenderer.create(<Heading theme={theme} />);
    expect(component).toBeDefined();
  });

  test('should be an Heading1 by default', () => {
    ReactTestUtils.isElementOfType(
      <Heading theme={theme} />,
      Heading1,
    );
  });

  test('should be an Heading with size', () => {
    ReactTestUtils.isElementOfType(
      <Heading theme={theme} size={3} />,
      Heading3,
    );
  });
});
