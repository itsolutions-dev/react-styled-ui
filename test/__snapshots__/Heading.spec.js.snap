// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render an Heading 1`] = `
.c0 {
  font-size: 28px;
  font-weight: 600;
  line-height: 30px;
  color: #4c4c4c;
}

<h1
  className="c0"
/>
`;
